### ISSUE TITLE
gunicorn fails to start if the pid does not have the same uid as expected

### ISSUE THREAD

**Issue by parak** [benoitc/gunicorn#1014]
In some rather rare cases, something like the following can happen:

1. gunicorn writes a pidfile as usual
2. The server dies unexpectedly
3. The server starts, and occupies up the pid written in 1 with some other non-gunicorn uid process (likely root).
4. gunicorn attempts to start, sees that the pid already exists, tries to kill it and errors out with "OSError: [Errno 1] Operation not permitted".
5. Deleting the pidfile allows gunicorn to start back up and recreate the pidfile with some new pid.

I'm a python newb (likely can be improved on by doing more checks), but something like the following workaround in pidfile.py works for me for now:

    elif e.args[0] == errno.EPERM:
        os.unlink(self.fname)
        return

**Comment by benoitc** [comment]
How can gunicorn can be restarted with a different user? You can use the [--user](http://docs.gunicorn.org/en/latest/settings.html#user) setting. In any case we shouldn't force the deletion of the pid there since it can be a issue in some cases also.

**Comment by parak** [comment]
Gunicorn is started normally on the system startup, with the same user as usual. However other processes will run before it (obviously), and one of them may take the same pid that gunicorn was using before the system crash. In my case it was a tty process, but it could be multiple things that start before gunicorn.

This can be easily replicated by stopping gunicorn (assuming it runs as a non privileged user), creating the pidfile with the pid of a root owned process, and attempting to start gunicorn.

Basically, I think that there needs to be some better way of handling such a corner case other than simply erroring out.

**Comment by tilgovi** [comment]
http://www.pathname.com/fhs/pub/fhs-2.3.html#VARRUNRUNTIMEVARIABLEDATA

We shouldn't have to worry about this. The directory where the pid file is stored is meant to only contain information written since boot.

Some distributions have this directory as a tmpfs mount for this reason.

**Comment by parak** [comment]
Ah, yep, you're right. I'll have to take better care with this on my end.

Thanks!
